spynnaker/pyNN/connections/spynnaker_poisson_control_connection.py

```diff
@@ -100,7 +101,8 @@ def _assemble_message(id_to_key_map, neuron_id_rates, pos):
         for _ in range(_MAX_RATES_PER_PACKET):
             neuron_id, rate = neuron_id_rates[pos]
             key = id_to_key_map[neuron_id]
-            message.add_key_and_payload(key, int(round(rate * scale)))
+            message.add_key_and_payload(
+                key, int(round(Decimal(str(rate)) * scale)))
```

Comment:
Is this necessary because `scale` is a `Decimal`, and therefore `rate` needs to be as well?  Seems a somewhat convoluted way of doing things, but if it's necessary then it's fine...

Reply:
This is an unfortunate consequence of using Decimal.  This is also how it is done in DataSpecification.